### PR TITLE
Refactor derived model

### PR DESCRIPTION
**Pull request by andy-beer**
derived_model.py replaced with a package of small files
yet to do:

- reduce complexity of some of the functions
- add unit tests for functions currently untested

Resolves issue #262

### CODE REVIEW

File: resqpy/derived_model/__init__.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+from .dm_extract_box import extract_box
+from .dm_fault_throw_scaling import fault_throw_scaling, global_fault_throw_scaling
+from .dm_gather_ensemble import gather_ensemble
+from .dm_interpolated_grid import interpolated_grid
+from .dm_local_depth_adjustment import local_depth_adjustment
+from .dm_refined_grid import refined_grid
+from .dm_tilted_grid import tilted_grid
+from .dm_unsplit_grid import unsplit_grid
+from .dm_zonal_grid import zonal_grid, single_layer_grid
+from .dm_zone_layer_ranges_from_array import zone_layer_ranges_from_array
```
Comment (emmanesbit) [inline resqpy/derived_model/__init__.py:29]: We don't seem to be importing dm_common.py - does this only contain functions used by others in this directory, not something users might call?

Comment (andy-beer) [reply resqpy/derived_model/__init__.py:29]: That's right, dm_common only holds 'private' functions used by some of the dm_* modules but not to be made available to calling code.

File: resqpy/derived_model/dm_zonal_grid.py [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+    model.h5_release()
+    if new_epc_file:
+        __write_grid(new_epc_file, grid, grid_title = new_grid_title, mode = 'w')
+    else:
+        __write_grid(epc_file, grid, ext_uuid = None, grid_title = new_grid_title, mode = 'a')
+
+    return grid
+
+
+def single_layer_grid(epc_file,
```
Comment (emmanesbit) [inline resqpy/derived_model/dm_zonal_grid.py:123]: Where is this used?

Comment (emmanesbit) [reply resqpy/derived_model/dm_zonal_grid.py:123]: Ah I see

**Review by emmanesbit — APPROVED** [review]
Looks good to me

**Pull request merged**